### DISCUSSION TITLE
PEM via App Store Connect API

### DISCUSSION THREAD

**Discussion by rdpcodes** [fastlane/fastlane#18333] [Q&A] [answered]
Hey,

I just wanted to ask if there are any plans to allow PEM to function via the AppStore Connect API? Does PEM currently not work because its not possible to do via the API or is it just because it hasn't been implemented yet?

<img width="632" alt="image" src="https://user-images.githubusercontent.com/12813087/109959093-9d33bc80-7d19-11eb-8392-d83066246bbc.png">

Thanks:)

**Comment by dinsen** [comment, marked as answer]
Hi @skyzadev 
There is no endpoints within the App Store Connect API.
See the topic Provisioning:
https://developer.apple.com/documentation/appstoreconnectapi

I can instead recommend generating an Auth Key and use that to send out push notifications. 
It will work for all apps on the developer account and will never expire.
https://developer.apple.com/documentation/usernotifications/setting_up_a_remote_notification_server/establishing_a_token-based_connection_to_apns
